src/routes/Home.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { v4 as uID } from 'uuid';
+import { useNavigate } from 'react-router';
+import { getCurrency } from '../redux/currency/currencySlice';
+import Currency from '../components/currency/Currency';
+import notFound from '../media/not-found.png';
+
+const Home = () => {
+  const fetchStatus = useRef(true);
+  const { currencyArr } = useSelector((state) => state.currency);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const [currencyWanted, setCurrencyWanted] = useState('');
+
+  useEffect(() => {
+    if (fetchStatus.current) {
+      fetchStatus.current = false;
+      dispatch(getCurrency());
+    }
+  }, []);
+
+  const checkClick = (item) => {
+    if (item.id) {
+      navigate(`details/${item.id}`);
+    }
+  };
+
+  const filterArray = currencyArr ? currencyArr.filter((item) => {
+    if (currencyWanted === '') {
+      return item;
+    }
+
+    if (item.name.toLowerCase().includes(currencyWanted.toLowerCase())) {
+      return item;
+    }
+
+    return null;
+  }) : [];
+
+  const result = () => {
+    if (filterArray.length === 0) {
+      return (
+        <div className="not-found">
+          <h3 className="not-foundTitle">CryptoCurrency not found.</h3>
+          <img src={notFound} alt="Not Found" className="not-foundImg" />
+        </div>
+      );
+    }
+    return filterArray.map((item) => (
+      <>
+        <div
+          key={uID()}
+          aria-hidden="true"
+          onClick={() => checkClick(item)}
+          className="currency-div"
+        >
+          <Currency
+            img={item.icon}
+            name={item.name}
+            price={item.price}
+            priceChange={item.priceChange1w}
+            symbol={item.symbol}
+          />
+        </div>
+      </>
+    ));
+  };
+
+  return (
+    <>
+      <h1 className="hero-title">CryptoCurrency Tracker (CCT)</h1>
+      <div className="input-container">
+        <input
+          type="text"
+          onChange={(e) => setCurrencyWanted(e.target.value)}
+          placeholder="Search for a crypto..."
+          value={currencyWanted}
+        />
+      </div>
+      <h3 className="currency-title darker-bg">Top 30 Cryptos</h3>
+      <div className="currency-container flex">{result()}</div>
+    </>
+  );
+};
+
```

Comment:
You have done a nice job creating the functionality of the application😉. But let's make a bit of improvement by sticking to the given design as this is a requirement in this milestone👍.
- just as it is shown in the design, let's make the `home` page have 2 items in each row. 

| Your design                                                             | Figma design                                         |
| --------------------------------------------------------------------------------| --------------------------------------------------------- |
|<img width="300" alt="image" src="https://github.com/hernandanielzamora/Stock-Data/assets/64643995/e179430b-f11f-4b47-a6ec-e8512de7705c">|<img width="300" alt="image" src="https://github.com/hernandanielzamora/Stock-Data/assets/64643995/337d76f7-e66a-4631-95e0-2b536cacf002">|

- Also, Kindly ensure to make use of the right font properties😉 The font-size and font-weight should look like the design as required👍.